source/common/http/codes.h

```diff
@@ -118,10 +86,31 @@ class CodeStatsImpl : public CodeStats {
   Stats::StatName vhost_;
   Stats::StatName zone_;
 
-  mutable RequestCodeGroup canary_upstream_rq_;
-  mutable RequestCodeGroup external_upstream_rq_;
-  mutable RequestCodeGroup internal_upstream_rq_;
-  mutable RequestCodeGroup upstream_rq_;
+  // Use an array of atomic pointers to hold StatNameStorage objects for
+  // every conceivable HTTP response code. In the hot-path we'll reference
+  // these with a null-check, and if we need to allocate a symbol for a
+  // new code, we'll take a mutex to avoid duplicate allocations and
+  // subsequent leaks. This is similar in principle to a ReaderMutexLock,
+  // but should be faster, as ReaderMutexLocks appear to be too expensive for
+  // fine-grained controls. Another option would be to use a lock per
+  // stat-name, which might have similar performance to atomics with default
+  // barrier policy.
+  //
+  // We don't allocate these all up front during construction because
+  // SymbolTable greedily encodes the first 128 names it discovers in one
+  // byte. We don't want those high-value single-byte codes to go to fully
+  // enumerating the 4 prefixes combined with HTTP codes that are seldom used,
+  // so we allocate these on demand.
+  //
+  // There can be multiple symbol tables in a server. The one passed into the
+  // Codes constructor should be the same as the one passed to
+  // Stats::ThreadLocalStore. Note that additional symbol tables can be created
+  // from IsolatedStoreImpl's default constructor.
+  //
+  // The Codes object is global to the server.
+
+  static constexpr uint32_t NumHttpCodes = 1000;
```

Comment:
Given that we don't crash now, just make this 600?

Reply:
Made it 500 and declared codes 0-99 invalid.